altapay.php

```diff
@@ -0,0 +1,2353 @@
+                );
+            } else {
+                $terminalArray[$terminal->getTitle()] = $terminal;
+            }
+        }
+
+        return $terminalArray;
+    }
+
+
+    /* ******************************** */
+
+    private function getAPIUsername()
+    {
+        return $this->username;
+    }
+
+    /* ******************************** */
+
+    private function getAPIPassword()
+    {
+        return $this->password;
+    }
+
+    /* ******************************** */
+
+    public function getAltapayUrl()
+    {
+        return $this->url;
+    } /* ******************************** */
+
+    /**
+     * Get field values for add/edit terminal form
+     *
+     * @return Array Associative array of object values
+     */
+    public function getFormValues()
+    {
+        $data = array();
+        $idTerminal = (int)Tools::getValue('id_terminal');
+
+
+        if ($idTerminal > 0) {
+            $data = new Terminal($idTerminal);
+        } else {
+            $def = Terminal::$definition;
+            foreach ($def['fields'] as $fieldName => $stuff) {
+                $data[$fieldName] = Tools::getValue($fieldName);
+            }
+        }
+
+        return $data;
+    }
+
+    /**
+     * Handle payment processing
+     * capture, refund, release
+     */
+    private function processPaymentActions()
+    {
+        // $cookie param is native prestashop framework param, being utilized in the frontend payment hook processing
+        global $cookie;
+        header('Content-Type: application/json');
+        if (!(Tools::getValue('action') && Tools::getValue('payment_id'))) {
+            return;
+        }
+
+        $paymentID = (int)Tools::getValue('payment_id');
+        // Merchant API
+        $api = new MerchantAPI();
+        try {
+            $api->init($this->getAltapayUrl(), $this->getAPIUsername(), $this->getAPIPassword());
+        } catch (Exception $e) {
+            saveLastErrorMessage($paymentID, $e->getMessage());
+
+            echo json_encode(
+                array(
+                    'status' => "error",
+                    'message' => "Connection error: " . $e->getMessage()
+                )
+            );
+            die();
+        }
+
+        $action = Tools::ucfirst(Tools::getValue('action'));
+        $goodWillRefund = false;
+        $orderID = Tools::getValue('ap_order_id');
+        $orderLines = Tools::getValue('ap_order_qty');
+        $orderLineGiftWrap = Tools::getValue('ap_order_wrap');
+        // CAPTURE
+        if ($action == 'Capture') {
+            try {
+                $finalOrderLines = $this->populateOrderLinesFromPost($orderLines, $orderLineGiftWrap, $orderID);
+                $api->captureAmount($paymentID, $finalOrderLines, Tools::getValue('amount'));
+                markAsCaptured($paymentID, $this->getItemCaptureRefundQuantityCount($finalOrderLines));
+            } catch (Exception $e) {
+                //save the latest error message in db
```

Comment:
Would be nice to have a consistent styling of comments:
```suggestion
                // Save the latest error message in db
```